book/src/pil/types.md

```diff
@@ -9,13 +9,77 @@ The powdr-pil language has the following types:
 - tuple
 - array
 - function type
-- expression
-- constraint
+- `expr` (expression)
+- `constr` (constraint)
+- `!` ("bottom" or "unreachable" type)
+
+> The `col` type is special in that it is only used for declaring columns, but cannot appear as the type of an expression.
+> See [Types of Symbols](#declaring-and-referencing-columns) for details.
```

Comment:
name of the section has changed

Reply:
Haha, I updated the link but forget the text :)